### PR TITLE
Family consent viewModel tests (EXPOSUREAPP-12545)

### PR DESCRIPTION
**Pull request by schauersbergern**
Add FamilyTestConsentViewModelTest

remarks:
for tests nameChanged and nameNotChanged viewModelDcc.isSubmittable.value remains null so these two tests fail,
maybe someone could point me in the right direction whats missing

### CODE REVIEW

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import org.junit.Assert
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import testhelpers.TestDispatcherProvider
+import testhelpers.extensions.InstantExecutorExtension
+
+@ExtendWith(InstantExecutorExtension::class)
+class FamilyTestConsentViewModelTest {
+
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt:24]: Please extend BaseTest()

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        viewModelDcc = FamilyTestConsentViewModel(
+            dispatcherProvider = TestDispatcherProvider(),
+            coronaTestQRCode = qrDcc,
+            familyTestCensor = familyTestCensor,
+            registrationStateProcessor = testRegistrationStateProcessor
+        )
+    }
+
+    @Test
+    fun nameChanged(value: String) {
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt:60]: what is the value for? this does not work

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        viewModelDcc.nameChanged("My Name")
+        viewModelDcc.isSubmittable.value shouldBe true
+    }
+
+    @Test
+    fun nameNotChanged(value: String) {
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt:66]: those two tests are identical, how would it result once in true and once in false?

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            coronaTestQRCode = qrDcc,
+            familyTestCensor = familyTestCensor,
+            registrationStateProcessor = testRegistrationStateProcessor
+        )
+    }
+
+    @Test
+    fun nameChanged(value: String) {
+        viewModelDcc.nameChanged("My Name")
+        viewModelDcc.isSubmittable.value shouldBe true
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt:62]: You need an observer for LiveData to emit values

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt (hunk 1 above)
Comment (schauersbergern) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt:25]: changed to extend BaseTest

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt (hunk 3 above)
Comment (schauersbergern) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModelTest.kt]: Adapted tests, was copy paste fauxpass

**Review by mtwalli — APPROVED** [review]
LGTM

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModel.kt [hunk 5]
```diff
@@ -29,7 +28,7 @@ class FamilyTestConsentViewModel @AssistedInject constructor(
 
     val isSubmittable = combine(personName, registrationStateProcessor.state) { name, state ->
         name.isNotEmpty() && state !is TestRegistrationStateProcessor.State.Working
-    }.asLiveData()
+    }.asLiveData2()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModel.kt:31]: @schauersbergern  FYI, this was the fix `asLiveData` was not using the dispatcher provider that you pass in the test 
either use pass it `asLiveData(dispatcher.Default)` or use `asLiveData2` which does it for you ;)

Comment (schauersbergern) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModel.kt:31]: @mtwalli Perfect, thanks for checking!

**Review by chiljamgossow — APPROVED** [review]

**Pull request merged**